### PR TITLE
Create Note Listing Page UI 

### LINKED ISSUE #1: Create note listing page UI

### PR DESCRIPTION
**Pull request by naufil50**
Fixes #1 

notes ui

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/NotesList.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Clock, MenuHorizontal } from "@bigbinary/neeto-icons";
+import { Typography, Tag, Dropdown, Avatar } from "@bigbinary/neetoui";
+
+export default function NotesList({ notes, index }) {
+  return (
+    <div className="mt-2 flex w-full flex-col">
+      {notes.map(note => (
+        <div
+          className="mb-4 w-full border border-gray-200 bg-white p-6 shadow-md dark:border-gray-700 dark:bg-gray-800"
+          key={index}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NotesList.jsx:12]: Please understand the purpose of having a `key` prop attached to all list elements. From the looks of how you have structured this component, the `index` argument will always have the same value, and that value has been passed to `key` prop, which would defeat the purpose of having the `key` prop in the first place.

Read:

- https://medium.com/wesionary-team/using-index-as-a-key-is-an-anti-pattern-in-react-8e5db3aea3a9
- https://reactjs.org/docs/lists-and-keys.html

File: app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+
+  return (
+    <div>
+      <MenuBar showMenu={showMenu} title="Notes">
+        <MenuBar.Block label="All" count={200} active />
+        <MenuBar.Block label="Users" count={80} />
+        <MenuBar.Block label="Leads" count={60} />
+        <MenuBar.Block label="Visitors" count={60} />
+
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx:17]: Please make sure to carefully read the `React Best Practices` section from the `Learn React` course in BBA.

https://academy.bigbinary.com/learn-react/pages/miscellaneous-tips#:~:text=Don%E2%80%99t%20add%20unnecessary%20blank%20lines

File: app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <MenuBar.SubTitle
+          iconProps={[
+            {
+              icon: Search,
+              onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx:22]: https://academy.bigbinary.com/learn-react/pages/know-when-to-use-previous-state-for-state-update

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 4]
```diff
@@ -1,102 +1,61 @@
...
-        <DeleteAlert
-          selectedNoteIds={selectedNoteIds}
-          onClose={() => setShowDeleteAlert(false)}
-          refetch={fetchNotes}
-          setSelectedNoteIds={setSelectedNoteIds}
+        <NotesList notes={notes} />
+        <NewNotePane
+          showPane={showNewNotePane}
+          setShowPane={setShowNewNotePane}
+          fetchNotes={notes}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:47]: How does it make sense to pass in `notes` as value to an argument named `fetchNotes` which denotes an action?

If you don't have any specific `fetchNotes` implemented in neetoUI, then just return `notes` from `fetchNotes` function. That's more sensible and the reader would easily understand what's going on.

**Review by yedhink — COMMENTED** [review]
@naufil50  I've added some comments.

- I have not pinpointed each and every spot where a similar change ought to be made. Thus when you work on a review comment, please make sure to check your whole codebase to see where a similar change needs to be made.
- You can make these changes as part of your next feature PR or raise a separate PR just fixing all the review changes.

File: app/javascript/src/components/Dashboard/Notes/NotesList.jsx (hunk 1 above)
Comment (naufil50) [reply app/javascript/src/components/Dashboard/Notes/NotesList.jsx:12]: This is updated afterwards but now have added id on object. Thanks for pointing this out

File: app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx (hunk 2 above)
Comment (naufil50) [reply app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx:17]: Done.Thanks

File: app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx (hunk 3 above)
Comment (naufil50) [reply app/javascript/src/components/Dashboard/Notes/NotesMenuBar.jsx:22]: This not needed for now. Thanks.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 4 above)
Comment (naufil50) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:47]: Yes, removed it now since we are using notes from constants. Thanks.

**Pull request merged**